blob/singularity_store.go

```diff
@@ -0,0 +1,151 @@
+package blob
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/data-preservation-programs/singularity/client"
+	"github.com/data-preservation-programs/singularity/handler/dataset"
+	"github.com/data-preservation-programs/singularity/handler/datasource"
+)
+
+const MOTION_DATASET_NAME = "MOTION_DATASET"
+const MAX_CAR_SIZE = "31.5GiB"
+
+type SingularityStore struct {
+	local             *LocalStore
+	sourceID          uint32
+	singularityClient client.Client
+}
+
+func NewSingularityStore(dir string, singularityClient client.Client) *SingularityStore {
+	local := NewLocalStore(dir)
+	return &SingularityStore{
+		local:             local,
+		singularityClient: singularityClient,
+	}
+}
+
+func (l *SingularityStore) Start(ctx context.Context) error {
+	_, err := l.singularityClient.CreateDataset(ctx, dataset.CreateRequest{
+		Name:       MOTION_DATASET_NAME,
+		MaxSizeStr: MAX_CAR_SIZE,
+	})
+	var asDuplicatedRecord client.DuplicateRecordError
+
+	// return errors, but ignore duplicated record, that means we just already created it
+	if err != nil && !errors.As(err, &asDuplicatedRecord) {
+		return err
+	}
+	source, err := l.singularityClient.CreateLocalSource(ctx, MOTION_DATASET_NAME, datasource.LocalRequest{
+		SourcePath:        l.local.dir,
+		RescanInterval:    "0",
+		DeleteAfterExport: false,
```

Comment:
This means stored files will remain in local directory indefinietly. Correct?

Reply:
for now yes.

Reply:
we will delete once retrieval is available. though possibly not through this flag.